pkg/controller/workload/jobframework/helpers.go

```diff
@@ -121,9 +288,20 @@ func EnsureOneWorkload(ctx context.Context, cli client.Client, req ctrl.Request,
 	return match, nil
 }
 
-// StartJob will unsuspend the job, and also inject the node affinity.
-func StartJob(ctx context.Context, client client.Client, record record.EventRecorder, job GenericJob, wl *kueue.Workload) error {
-	nodeSelectors, err := GetNodeSelectors(ctx, client, wl)
+// equivalentToWorkload checks if the job corresponds to the workload
+func (r *GenericJobReconciler) equivalentToWorkload(job GenericJob, wl *kueue.Workload) bool {
+	owner := metav1.GetControllerOf(wl)
+	// Indexes don't work in unit tests, so we explicitly check for the
+	// owner here.
+	if owner.Name != job.Object().GetName() {
```

Comment:
This is duplicated in `EnsureOneWorkload`, I think we can avoid this like https://github.com/kubernetes-sigs/kueue/blob/48156ad74e2cb4598076fdab747789cacdca23ed/pkg/util/testing/client.go#L42-L43

Reply:
Removed the duplication by deleting the code in front. We could probably replace it by using unit test `WithIndex` as you point out, but this would be out of scope for this PR, IMO.